onnxruntime/core/providers/xnnpack/xnnpack_init.h

```diff
@@ -0,0 +1,11 @@
+#include "core/framework/allocator.h"
+
+struct xnn_allocator;
+
+namespace onnxruntime {
+namespace xnnpack {
+
+std::pair<AllocatorPtr*, xnn_allocator*> GetStoredAllocator();
+
```

Comment:
nit: As AllocatorPtr is a shared_ptr, having a pointer to a shared_ptr seems like it adds an unnecessary level of indirection. Any reason this can't simply return AllocatorPtr in the pair?

Reply:
`reference_wrapper` seems even ugly, I'd rather return a pointer of shared_ptr.
 
I need this static shared_ptr to increase ref and keep it valid.  A returned `AllocatorPtr` is just a new and independent shared_ptr and it will be ending with resource freeing. 

1. we need a resident object which keep Allocator alive.
2. we need to store the temp Allocator to resident AllocatorPtr.

Please correct me if It's wrong.

Reply:
Ah you're right - was thinking about a non-empty shared_ptr but that doesn't help when it's being assigned to.

I think it's possible to just return a reference in the std::pair though. This example unit test passes and returns a reference in the pair. I _think_ that should be ok for our usage. 

```c++
static std::pair<std::shared_ptr<std::string>&, int> GetShared() {
  static std::shared_ptr<std::string> shared;
  return {shared, 0};
}

TEST(TempTest, SharedPtrUpdates) {
  const auto& [first, count] = GetShared();
  auto local = std::make_shared<std::string>("set");
  if (!first) {
    first = local;
  }

  const auto& [second, count2] = GetShared();
  ASSERT_TRUE(second != nullptr) << "Shared pointer should have been set.";
  ASSERT_EQ(*second, "set");
}
```

Reply:
Sounds good.

It should work for our scenario if test case passed.